### PR TITLE
Seed RNG with sample data on Piksi v3.

### PR DESCRIPTION
**Pull request by gsmcmullin**
/cc @swift-nav/firmware

### CODE REVIEW

File: src/board/v3/init.c [hunk 1]
```diff
@@ -78,7 +91,7 @@ void init(void)
   nap_auth_check();
   nap_callbacks_setup();
 
-  srand(0);
+  random_init();
```
Comment (jacobmcnamee) [inline src/board/v3/init.c]: Need to initialize frontend before trying to read out samples.

File: src/board/v3/init.c [hunk 2]
```diff
@@ -68,6 +69,18 @@ void pre_init(void)
   led_setup();
 }
 
+static void random_init(void)
+{
+  u64 sample_data;
+  u32 seed = 0;
+
+  raw_samples_get(&sample_data, sizeof(sample_data));
```
Comment (jacobmcnamee) [inline src/board/v3/init.c]: Log error if `raw_samples_get()` returns false.

**Comment by jacobmcnamee** [comment]
@wltr What bits of the samples are actually "random"? There is now a counter in the upper nibble right?

**Comment by jacobmcnamee** [comment]
See #689

File: src/board/v3/init.c (hunk 1 above)
Comment (gsmcmullin) [inline src/board/v3/init.c]: This happens in `frontend_configure` called from `nap_setup` above.

**Comment by wltr** [comment]
@jacobmcnamee The 4 LSBs are random. The counter was definitely needed for the bare-metal sample grabber firmware, but otherwise I don't think so. However, I suppose it can't hurt to be able to make sure that you get a continuous stream.

**Comment by jacobmcnamee** [comment]
Can you delete https://github.com/swift-nav/piksi_firmware/blob/master/src/board/v3/frontend.c#L155 since `frontend_configure()` is now called in v3 `init()` ?

**Pull request closed without merging**